stage0/src/main.rs

```diff
@@ -112,6 +112,62 @@ pub unsafe fn jump_to_kernel(entry_point: VirtAddr) -> ! {
     );
 }
 
+/// Trait for abstracting the call to `PVALIDATE` over different page sizes and ranges.
+trait Validator {
+    fn pvalidate(self, validated: Validation) -> Result<(), InstructionError>;
+}
+
+impl Validator for PhysFrame<Size2MiB> {
+    fn pvalidate(self, validated: Validation) -> Result<(), InstructionError> {
```

Comment:
I'll defer to @conradgrobler here; the `pvalidate()` function in `sev_guest` names that parameter `validated`, so I'm inclined to keep it the same.

Reply:
I agree this is a bit inconsistent. IIRC the parameter in the sev_guest crate started out as `validated: bool` and then later became an enum, and I forgot to rename the parameter when the type changed. A more accurate naming would be something like `validation_state: ValidationState`. It would be nice to make it consistent, but we can also do that in a future cleanup PR.

Reply:
There's nothing more permanent than a temporary solution, but fine, let's punt it :)